src/lotus/client.rs

```diff
@@ -500,4 +640,105 @@ impl LotusJsonRPCClient<JsonRpcClientImpl> {
         let jsonrpc_client = JsonRpcClientImpl::new(url, auth_token);
         LotusJsonRPCClient::new(jsonrpc_client)
     }
+
+    pub fn from_subnet_with_wallet_store(
+        subnet: &crate::config::Subnet,
+        wallet_store: Arc<RwLock<Wallet>>,
+    ) -> Self {
+        let url = subnet.jsonrpc_api_http.clone();
+        let auth_token = subnet.auth_token.as_deref();
+        let jsonrpc_client = JsonRpcClientImpl::new(url, auth_token);
+        LotusJsonRPCClient::new_with_wallet_store(jsonrpc_client, wallet_store)
+    }
+}
+
+fn create_signed_message_params(msg: MpoolPushMessage, signature: Signature) -> serde_json::Value {
+    let nonce = msg
+        .nonce
+        .map(|n| serde_json::Value::Number(n.into()))
+        .unwrap_or(serde_json::Value::Null);
+
+    let to_value_str = |t: Option<TokenAmount>| {
+        t.map(|n| serde_json::Value::String(n.atto().to_u64().unwrap().to_string()))
+            .unwrap_or(serde_json::Value::Null)
+    };
+    let to_value = |t: Option<TokenAmount>| {
+        t.map(|n| serde_json::Value::Number(n.atto().to_u64().unwrap().into()))
+            .unwrap_or(serde_json::Value::Null)
+    };
+
+    let gas_limit = to_value(msg.gas_limit);
+    let gas_premium = to_value_str(msg.gas_premium);
+    let gas_fee_cap = to_value_str(msg.gas_fee_cap);
+
+    let Signature { sig_type, bytes } = signature;
+    let sig_encoded = base64::engine::general_purpose::STANDARD.encode(bytes);
+
+    let params_encoded = base64::engine::general_purpose::STANDARD.encode(msg.params);
+    // refer to: https://lotus.filecoin.io/reference/lotus/mpool/#mpoolpush
+    json!([
+        {
+            "Message": {
+                "Version": msg.version.unwrap_or(0),
+                "To": msg.to.to_string(),
+                "From": msg.from.to_string(),
+                "Value": msg.value.atto().to_string(),
+                "Method": msg.method,
+                "Params": params_encoded,
+
+                // THESE ALL WILL AUTO POPULATE if null
+                "Nonce": nonce,
+                "GasLimit": gas_limit,
+                "GasFeeCap": gas_fee_cap,
+                "GasPremium": gas_premium,
+                "CID": CIDMap::from(msg.cid),
+            },
+            "Signature": {
+                "Type": sig_type as u8,
+                "Data": sig_encoded,
+            }
+        }
+    ])
 }
+
+// #[cfg(test)]
```

Comment:
Should we remove this one or do you want to keep it for reference in some form? If this is the case we should consider making it cleaner.

Reply:
Removed